### PR TITLE
feature: Add business details and hyper parameters fields and update test_model_card.py

### PR DESCRIPTION
**Pull request by Andywangnn**
*Issue #, if available:*

*Description of changes:*
feature: add business details and hyper parameters fields to python SDK.

*Testing done:*
Passed unit test for test_model_card.py by running  export IGNORE_COVERAGE=- ; tox   -- -s -vv tests/unit/test_model_card.py ; unset IGNORE_COVERAGE  

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/model_card/model_card.py [hunk 1]
```diff
@@ -363,6 +386,27 @@ def __init__(
         self.notes = notes
 
 
+class HyperParameter(_DefaultToRequestDict, _DefaultFromDict):
+    """Hyper-Parameters data.
+
+    Should only be used during auto-population of parameters details.
```
[the remark this answers is not in the document]
Comment (Andywangnn) [reply src/sagemaker/model_card/model_card.py:392]: > why `Should only be used during auto-population of parameters details.` ?

This `Should only be used during auto-population of parameters details.`is Wrong comment, will delete it.

File: tests/unit/test_model_card.py [hunk 2]
```diff
@@ -866,6 +891,14 @@ def __init__(self, attr1):  # pylint: disable=C0116
     ):
         ExampleClass(attr1=IntendedUses())
 
+    with pytest.raises(
```
[the remark this answers is not in the document]
Comment (Andywangnn) [reply tests/unit/test_model_card.py:894]: > I don't think we need this lines. The goal of this test is to see if `_IsModelCardObject` descriptor working as expected and not specific to IntendedUses class

make sense, will remove those lines for business details

File: tests/unit/test_model_card.py [hunk 3]
```diff
@@ -1017,6 +1065,9 @@ def test_training_details_autodiscovery_from_model_overview(
     assert len(training_details.training_job_details.training_metrics) == len(
         SEARCH_TRAINING_JOB_EXAMPLE["Results"][0]["TrainingJob"]["FinalMetricDataList"]
     )
+    assert len(training_details.training_job_details.training_metrics) == len(
```
Comment (Zhenshan-Jin) [inline tests/unit/test_model_card.py:1068]: `training_details.training_job_details.training_metrics` ? we should assert hyper_paramters

File: tests/unit/test_model_card.py [hunk 4]
```diff
@@ -1063,6 +1115,9 @@ def test_training_details_autodiscovery_from_job_name(session):
...
```
Comment (Zhenshan-Jin) [inline tests/unit/test_model_card.py:1118]: why `training_details.training_job_details.training_metrics` ?

File: tests/unit/test_model_card.py [hunk 5]
```diff
@@ -1047,6 +1080,7 @@ def test_training_details_autodiscovery_from_model_overview_autopilot(
     )
 
     assert len(training_details.training_job_details.training_metrics) == 0
+    assert len(training_details.training_job_details.hyper_parameters) == 0
```
[the remark this answers is not in the document]
Comment (Zhenshan-Jin) [reply tests/unit/test_model_card.py:1101]: please comment your fix if any

File: tests/unit/test_model_card.py (hunk 3 above)
Comment (Andywangnn) [reply tests/unit/test_model_card.py:1068]: Will change training_metrics to hyper_parameters.

File: tests/unit/test_model_card.py (hunk 4 above)
Comment (Andywangnn) [reply tests/unit/test_model_card.py:1118]: Will change training_metrics to hyper_parameters.

File: tests/unit/test_model_card.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Zhenshan-Jin) [reply tests/unit/test_model_card.py:1101]: please comment your fix if any

File: src/sagemaker/model_card/model_card.py [hunk 6]
```diff
@@ -442,6 +493,11 @@ def _create_training_details(training_job_data: dict, cls: "TrainingDetails", **
                 ]
                 if "FinalMetricDataList" in training_job_data
                 else [],
+                # "hyper_parameters": print(training_job_data["HyperParameters"]),
```
Comment (Zhenshan-Jin) [inline src/sagemaker/model_card/model_card.py:496]: remove this line

File: tests/unit/test_model_card.py [hunk 7]
```diff
@@ -1046,7 +1090,10 @@ def test_training_details_autodiscovery_from_model_overview_autopilot(
         model_overview=model_overview_example, sagemaker_session=session
     )
 
+    # There are 0 required keys in training_metrics field in SEARCH_TRAINING_JOB_AUTOPILOT_EXAMPLE has
```
Comment (Zhenshan-Jin) [inline tests/unit/test_model_card.py:1093]: It looks confusing. If you want, you could just add why `SEARCH_TRAINING_JOB_AUTOPILOT_EXAMPLE` result in this issue: `MetricDefinitions` is empty.

File: tests/unit/test_model_card.py [hunk 8]
```diff
@@ -1046,7 +1090,10 @@ def test_training_details_autodiscovery_from_model_overview_autopilot(
...
     assert len(training_details.training_job_details.training_metrics) == 0
+    # There are 3 required keys in hyper_parameters field in SEARCH_TRAINING_JOB_AUTOPILOT_EXAMPLE has
```
Comment (Zhenshan-Jin) [inline tests/unit/test_model_card.py:1095]: same as above

File: src/sagemaker/model_card/model_card.py (hunk 6 above)
Comment (Andywangnn) [reply src/sagemaker/model_card/model_card.py:496]: removed

**Review by Zhenshan-Jin — APPROVED** [review]

**Review by vikas0203 — COMMENTED** [review]
/bot run all

**Review by vikas0203 — COMMENTED** [review]
/bot run pr

**Review by vikas0203 — APPROVED** [review]

**Pull request merged**